commons/src/main/java/org/infinispan/commons/marshall/SerializeWith.java

```diff
@@ -33,4 +33,10 @@
     */
    Class<? extends Externalizer<?>> value();
 
+   /**
+    * TODO
+    * @return
+    */
+   SerializeValueMatcher valueMatcher() default SerializeValueMatcher.MATCH_ALWAYS;
```

Comment:
I know, this was just a quick fix. I think we should have a separate annotation to optionally define this. WDYT? Also, think about it in the context of having a way to define it along or instead, the function to execute when retrying?